MaskedSelect.lua

```diff
@@ -0,0 +1,69 @@
+local MaskedSelect, parent = torch.class('nn.MaskedSelect', 'nn.Module')
+
+--[[ Sets the provided mask value for the module. ]]
+function MaskedSelect:__init()
+  parent.__init(self)
+  self._maskIndices = torch.Tensor()
+  self._maskIndexBuffer = torch.Tensor()
+  self._gradBuffer = torch.Tensor()
+  self._gradMask = torch.Tensor()
+end
+
+--[[ Performs maskedSelect operation. ]]
+function MaskedSelect:updateOutput(input)
+  local input, mask = unpack(input)
+  self.output:maskedSelect(input, mask)
+  return self.output
+end
+
+--[[ Reverse maps unmasked gradOutput back to gradInput. ]]
+function MaskedSelect:updateGradInput(input, gradOutput)
+  local input, mask = unpack(input)
+  self._maskIndexBuffer:range(1, mask:nElement()):resize(mask:size())
```

Comment:
Currently, there is no `range` function in `cutorch`, meaning that this line won't work as is. A workaround would be to have another buffer in the CPU which computes the `range`, and then copy it to CPU.